x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java

```diff
@@ -992,35 +977,26 @@ private void findTokenFromRefreshToken(String refreshToken, Iterator<TimeValue>
             );
             findTokenFromRefreshToken(refreshToken, securityMainIndex, backoff, listener);
         } else {
-            if (refreshToken.length() == HASHED_TOKEN_LENGTH) {
-                logger.debug("Assuming a hashed refresh token [{}] retrieved from the tokens index", refreshToken);
-                findTokenFromRefreshToken(refreshToken, securityTokensIndex, backoff, listener);
```

Comment:
And this is the motivation for the change:
The interface of the `findTokenFromRefreshToken` method was complicated because it accepted both "user-facing" refresh tokens (i.e. unhashed) as well as "internal-only" ones (i.e. hashed), and it differentiated between the two by length...
But with the change at https://github.com/elastic/elasticsearch/pull/97422/files#r1254553360 , "internal-only" ones are not a valid input anymore, so this length check can be discarded.